dist/tools/git/git-cache

```diff
@@ -187,7 +188,23 @@ clone() {
         if [ -n "$tag" ]; then
             echo "git-cache: cloning from cache. tag=$tag"
             git -c advice.detachedHead=false clone $Q --reference "${GIT_CACHE_DIR}" --shared "${GIT_CACHE_DIR}" "${TARGET_PATH}" --branch $tag
+
+            # rename tags from <remote-hash>/* to *
             git -C "${TARGET_PATH}" fetch $Q origin "refs/tags/${REMOTE_NAME}/*:refs/tags/*"
+
+            # remove all commit* and <remote-hash>/* tags
+            git -C "${TARGET_PATH}" tag -l \
+                | grep -P '(^[a-f0-9]{40}/|^commit[a-f0-9]{40}(-\d+)?$)' \
+                | xargs git -C "${TARGET_PATH}" tag -d > /dev/null
```

Comment:
This looks a bit like the `cleanup()` function you introduced below (except for the `grep` pattern). Maybe unify?

Reply:
> (29 lines of bash script changes I do not consider a low hanging fruit btw)

Yeah, sorry. I thought this could count as a version bump (this is copied directly from upstream).

Reply:
postpone? as is, it is well tested.

Reply:
> postpone? as is, it is well tested.

So can't you run the tests over the updated version?

Reply:
well, testing is manual. honestly, I don't have the time.
the first term of the pipe cannot be easily factored out, as one is using "git_cache" instead of "git -Cpath". the two others could be factored into ```delete_tags "$regex" "$git-path"```.
I'm not sure readability would gain much.

Reply:
Ok.